lib/cucumber/formatter/junit.rb

```diff
@@ -157,6 +160,21 @@ def build_testcase(result, scenario_designation, output)
         @current_feature_data[:tests] += 1
       end
 
+      def get_testcase_attributes(classname, name, duration, filename)
+        attributes = { classname: classname, name: name, time: format('%<duration>.6f', duration: duration) }
+        attributes[:file] = filename if add_fileattribute?
+
+        attributes
+      end
+
+      def add_fileattribute?
+        return false if @config.formats.nil? || @config.formats.empty?
+
+        !!@config.formats.find do |format|
+          format[0] == 'junit' and format.dig(1, 'fileattribute') == 'true'
```

Comment:
Ideally should use `&&` here. Can also do `format.first` if it's enumerable (Which it looks like).

Reply:
It is. The formats come in as an array of arrays. Adjusted. Thanks.